static/app/components/searchQueryBuilder/tokens/combobox.tsx

```diff
@@ -439,20 +442,32 @@ export function SearchQueryBuilderCombobox<
       },
       onKeyDown: e => {
         onKeyDown?.(e, {state});
-        switch (e.key) {
-          case 'Escape':
-            state.close();
-            onExit?.();
-            return;
-          case 'Enter':
-            if (isOpen && state.selectionManager.focusedKey) {
-              return;
-            }
-            state.close();
-            onCustomValueCommitted(inputValue);
-            return;
-          default:
+
+        if (e.key === 'Escape') {
+          state.close();
+          onExit?.();
+          return;
+        }
+
+        if (e.key === 'Enter') {
+          if (isOpen && state.selectionManager.focusedKey) {
             return;
+          }
+          state.close();
+          onCustomValueCommitted(inputValue);
+          return;
+        }
+
+        // we need to force this open for aggregate filters e.g. count_if()
```

Comment:
Can you expand on this? I'm a bit confused at why this is a special case.

If it's specific to aggregate filters, would it be better to put the logic in the `onKeyDown` in that component instead?